atomic_reactor/plugins/exit_verify_media_types.py

```diff
@@ -59,77 +62,85 @@ def run(self):
-            if check_digests:
-                digests = get_manifest_digests(pullspec, registry_name, insecure, secret,
-                                               require_digest=False)
-                if digests:
-                    if digests.v2_list:
-                        self.log.info("Manifest list found")
-                        if MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST in expected_media_types:
-                            media_types.add(MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST)
-                    if digests.v2:
-                        self.log.info("V2 schema 2 digest found")
-                        if MEDIA_TYPE_DOCKER_V2_SCHEMA2 in expected_media_types:
-                            media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA2)
-                    if digests.v1:
-                        self.log.info("V2 schema 1 digest found")
-                        if MEDIA_TYPE_DOCKER_V2_SCHEMA1 in expected_media_types:
-                            media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA1)
-
-            if MEDIA_TYPE_DOCKER_V1 in expected_media_types:
-                if verify_v1_image(pullspec, registry_name, self.log, insecure, secret):
-                    media_types.add(MEDIA_TYPE_DOCKER_V1)
-
-            # sorting the media type here so the failure message is predictable for unit tests
-            missing_types = []
-            for media_type in sorted(expected_media_types):
-                if media_type not in media_types:
-                    missing_types.append(media_type)
-            if missing_types:
-                raise KeyError("expected media types {0} ".format(missing_types) +
-                               "not in available media types {0}".format(sorted(media_types)))
-        return sorted(media_types)
-
-    def set_manifest_list_expectations(self, expected_media_types):
+            digests = get_manifest_digests(pullspec, registry_name, insecure, secret,
+                                           require_digest=False)
+            if digests:
+                if digests.v2_list:
+                    media_types.add(MEDIA_TYPE_DOCKER_V2_MANIFEST_LIST)
+                if digests.v2:
+                    media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA2)
+                if digests.v1:
+                    media_types.add(MEDIA_TYPE_DOCKER_V2_SCHEMA1)
+                if digests.oci:
+                    media_types.add(MEDIA_TYPE_OCI_V1)
+                if digests.oci_index:
+                    media_types.add(MEDIA_TYPE_OCI_V1_INDEX)
+
+            if verify_v1_image(pullspec, registry_name, self.log, insecure, secret):
+                media_types.add(MEDIA_TYPE_DOCKER_V1)
+
+            media_in_registry[registry_name]['found'] = media_types
+
+        should_raise = False
+        all_found = set()
+        for registry_name, manifests in media_in_registry.items():
+            all_found.update(manifests['found'])
+            if manifests['expected'] - manifests['found']:
```

Comment:
Is it possible that 'found' contains some unexpected entry? If so, should we also capture the occurrence and log it?

Reply:
in 'found' we story only media_types which can be in 'expected_media_types'
if 'found' has more entries than 'expected' they will be returned, unless manifest list expectation only is met

we are treating 'expected_media_types' more like required types

Reply:
Thanks for the answer. I wanted to check if logging those would be useful at all!